### ISSUE TITLE
run_qwen3_4b_fsdp.sh token budgets left the GPU idle for a third of the actor update when training Qwen3.5-9B with GRPO on 8x B200

### ISSUE THREAD

**Issue by TarzanZhao** [verl-project/verl#7847]
### System Info

verl `main` at `89dad2d7`, editable install from the repo lockfile (`uv run --frozen --all-packages --extra vllm --extra fsdp`). Python 3.12.14, torch 2.11.0+cu130, vLLM 0.24.0, flash-attn 2.8.3, transformers 5.9.0, ray 2.55.1. One node, 8 x NVIDIA B200 (183 GB), driver 580.126.20, Linux.

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (GRPO on GSM8K)
- [ ] My own task or dataset

### Reproduction

The script as in the repo, with the step count, validation, checkpoints and seeds overridden (full command, environment and profiler settings are under Reproduce in #7833):

```
export MODEL_PATH=Qwen/Qwen3.5-9B NGPUS_PER_NODE=8 INFER_BACKEND=vllm SAVE_FREQ=-1 TEST_FREQ=-1
bash examples/grpo_trainer/run_qwen3_4b_fsdp.sh trainer.total_training_steps=6 trainer.val_before_train=False \
    data.seed=42 actor_rollout_ref.rollout.seed=42 actor_rollout_ref.actor.data_loader_seed=42
```

I ran the repo's GRPO on GSM8K example (`examples/grpo_trainer/run_qwen3_4b_fsdp.sh` with `MODEL_PATH=Qwen/Qwen3.5-9B`, FSDP actor and ref with a vLLM rollout, 8 GPUs) on 8x B200, following the repo's lockfile install, and got 151 s per training step. A profile of the actor update showed each mini-batch split into 62 micro-batches of about 2.7k tokens, from the token budgets in `examples/grpo_trainer/run_qwen3_4b_fsdp.sh` (`ppo_max_token_len_per_gpu=3000`, with 4096 and 8192 for the two log-prob stages). Under FSDP every micro-batch re-gathered the full 9B parameters and reduce-scattered fp32 gradients, so NCCL took longer than the GEMMs and the GPU sat idle for a third of the update, while the actor peaked at 34 GB of 183. With the three budgets raised to 16384, the same step took 110 s, for 5.8 GB more memory per GPU.

I opened #7833 to record my experiment: setup, measurements, traces and some potential fixes. A different fix may also well suit the codebase.

### Expected behavior

Nothing in the logs said how many micro-batches a mini-batch had been split into, and `docs/perf/perf_tuning.rst` set the budget from the sequence length alone (at least 2 x (max_prompt_length + max_response_length), 4096 here), so I only found the split by profiling. A log line per stage with the micro-batch count and the tokens per micro-batch, or a note in `docs/perf/perf_tuning.rst` tying the budget to the memory left after the model's footprint, would have saved me the profile.

**Comment by TarzanZhao** [comment]
Closing along with #7833; the measurements are kept there for reference.
